### PR TITLE
[widgets.mpd] Add fields `${Artists}` and `${Genres}`

### PR DESCRIPTION
**Pull request by toniz4**
Resolves GH-116 

Adds new return values to widgets.mpd that concatenate all artists and generes when there's more than one. For tracks with vorbis metadata.

I've also updated the docs to reflect those changes.

### CODE REVIEW

**Review by McSinyx — CHANGES_REQUESTED** [review]
The patchset looks good to me, aside from some minor formatting issues.  After you're done resolving them, could you please test if both `Artists` and `Genres` are working?

File: widgets/mpd_all.lua [hunk 1]
```diff
@@ -144,8 +151,25 @@ function mpd_all.async(format, warg, callback)
...
+                            mpd_state["{Artists}"] = append_with_separator(
+                                current_artists, v)
+                        end
+                    end
+                    if k == "Genre" then
+                        local current_generes = mpd_state["{Generes}"]
+                        if current_generes == "N/A" then
+                            mpd_state["{Generes}"] = v
+                        else
+                            mpd_state["{Generes}"] = append_with_separator(
```
[the remark this answers is not in the document]
Comment (toniz4) [reply widgets/mpd_all.lua]: Done

**Comment by toniz4** [comment]
> The patchset looks good to me, aside from some minor formatting issues. After you're done resolving them, could you please test if both `Artists` and `Genres` are working?

Genres was broken before because of the typo, but now It's working fine:

Here's the output:
image

The update function:
```lua
local update = function (widget, args)
   if args["{state}"] == "Stop" then
      widget.visible = false
      return ''
   else
      widget.visible = true
      return {
          ('Artists: %s Track: %s Genres: %s'):format(args["{Artists}"], args["{Title}"], args["{Genres}"]),
         args["{state}"]
      }
   end
end
```

And what mpd is returning:

```
$ printf 'password ""\nstatus\ncurrentsong\nclose \n' | curl --connect-timeout 1 -fsm 3 telnet://127.0.0.1:6600
OK MPD 0.23.5
ACK [3@0] {password} incorrect password
volume: 100
repeat: 0
random: 0
single: 0
consume: 0
partition: default
playlist: 1537
playlistlength: 192
mixrampdb: 0
state: play
song: 182
songid: 1556
time: 284:321
elapsed: 284.284
bitrate: 1221
duration: 320.625
audio: 44100:16:2
nextsong: 183
nextsongid: 1557
OK
file: Parannoul/Parannoul - To See the Next Part of the Dream/01 - Beautiful World.flac
Last-Modified: 2023-02-20T06:28:07Z
Format: 44100:16:2
Title: Beautiful World
Artist: Parannoul
Album: To See the Next Part of the Dream
AlbumArtist: Parannoul
Track: 1
Disc: 1
Genre: Asian Music
Genre: K-Pop
Date: 2021-04-19
Time: 321
duration: 320.625
Pos: 182
Id: 1556
OK
```

More than one artist it's working as expected too:

image

Here I couldn't find any tracks that have multiples of other tags, not being genre, artist or albumArtist.

**Review by McSinyx — APPROVED** [review]
Thanks, I will merge and tag a new release this weekend.

**Pull request merged**